pkg/storer/internal/reserve/reserve_test.go

```diff
@@ -752,6 +753,85 @@ func TestIterate(t *testing.T) {
 	})
 }
 
+func TestReset(t *testing.T) {
+	t.Parallel()
+
+	baseAddr := swarm.RandAddress(t)
+
+	ts := internal.NewInmemStorage()
+
+	r, err := reserve.New(
+		baseAddr,
+		ts,
+		0, kademlia.NewTopologyDriver(),
+		log.Noop,
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var chs []swarm.Chunk
+
+	for b := 0; b < 5; b++ {
+		for i := 1; i <= 50; i++ {
+			ch := chunk.GenerateTestRandomChunkAt(t, baseAddr, b)
+			err := r.Put(context.Background(), ch)
+			if err != nil {
+				t.Fatal(err)
+			}
+			stampHash, err := ch.Stamp().Hash()
+			if err != nil {
+				t.Fatal(err)
+			}
+			checkStore(t, ts.IndexStore(), &reserve.BatchRadiusItem{Bin: uint8(b), BatchID: ch.Stamp().BatchID(), Address: ch.Address(), StampHash: stampHash}, false)
+			checkStore(t, ts.IndexStore(), &reserve.ChunkBinItem{Bin: uint8(b), BinID: uint64(i), StampHash: stampHash}, false)
+			checkChunk(t, ts, ch, false)
+			_, err = r.Get(context.Background(), ch.Address(), ch.Stamp().BatchID(), stampHash)
+			if err != nil {
+				t.Fatal(err)
+			}
+			chs = append(chs, ch)
+		}
+	}
+
+	err = r.Reset(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	c, err := ts.IndexStore().Count(&reserve.BatchRadiusItem{})
```

Comment:
it could be checked before reset as well to show the expected count

Reply:
yes, good idea